src/content/docs/en/reference/container-reference.mdx

```diff
@@ -0,0 +1,292 @@
+
+## `renderToResponse()`
+
+It renders a component, and it returns a `Response` object.
+
+```js
+import { experimental_AstroContainer } from "astro/container";
+import Card from "../src/components/Card.astro";
+
+const container = await experimental_AstroContainer.create();
+const result = await container.renderToResponse(Card);
+```
+
+It accepts a [number of options](#rendering-options).
+
+### Rendering options
+
+Both [`renderToResponse`](#rendertoresponse) and [`renderToString`](#rendertostring) accept an object as their second argument:
+
+```ts
+export type ContainerRenderOptions = {
+	slots?: Record<string, any>;
+	request?: Request;
+	params?: Record<string, string | undefined>;
+	locals?: App.Locals;
+	routeType?: "page" | "endpoint";
+  
+};
+```
+
+#### `slots`
+
+**Type**: `Record<string, any>`;
+
+Use this options if your component needs to render some [slots](en/basics/astro-components/#slots).
+
+If your components renders one slot, pass an object with `default` as key:
+
+```js name="Card.test.js"
+import Card from "../src/components/Card.astro";
+
+const result = await container.renderToString(Card, { 
+  slots: { default: "Some value"}
+});
+```
+
+If your component renders named slots, use the keys of the object to name the slots:
+
+```astro name="Card.astro"
+---
+---
+<div>
+  <slot name="header" />
+  <slot name="footer" />
+</div>
+```
+
+```js name="Card.test.js"
+import Card from "../src/components/Card.astro";
+
+const result = await container.renderToString(Card, { 
+  slots: { "header": "Header content", "footer": "Footer" }
+});
+```
+
+You can also render components in cascade:
+
+```astro name="Card.astro"
+---
+---
+<div>
+  <slot name="header" />
+  <slot name="footer" />
+</div>
+```
+
+```js name="Card.test.js"
+import Card from "../src/components/Card.astro";
+import CardHeader from "../src/components/CardHeader.astro";
+import CardFooter from "../src/components/CardFooter.astro";
+
+const result = await container.renderToString(Card, { 
+  slots: { 
+    "header": await container.renderToString(CardHeader), 
+    "footer":  await container.renderToString(CardFooter),
+  }
+});
+```
+
+#### `request` option
+
+**Type**: `Request`
+
+The request is used to understand which path/URL the component is about to render.
+
+Use this option in case your component needs to read information like `Astro.url` or `Astro.request`.
+
+You can also inject possible headers or cookies.
+
+```js file="Card.test.js"
+import Card from "../src/components/Card.astro";
+
+const result = await container.renderToString(Card, { 
+  request: new Request("https://example.com/blog", {
+    headers: {
+      "X-some-secret-header": "test-value"
+    }
+  })
+});
+```
+
+#### `params` option
+
+**Type**: `Record<string, string | undefined>`;
+
+Use this option in case your component needs to read `Astro.params`. As opposed to `getStaticPaths`, you must provide only **one** item of the params returned by that function.
```

Comment:
```suggestion
An object to pass information about the path parameter to an Astro component responsible for [generating dynamic routes](/en/guides/routing/#dynamic-routes).

Use this option when your component needs a value for `Astro.params` in order to generate a single route dynamically.
```
Check this for accuracy, but I think this seems like what you're saying. 

And, I thought mentioning `getStaticPaths()` *might* be more confusing here than helpful? I think we can say this like above to emphasize that, like in other examples shown, you're providing *ONE* value of something so that you can test a particular case by rendering a single version of the component.